ImageD11/frelon_peaksearch.py

```diff
@@ -360,6 +360,100 @@ def guess_bg(ds, scan_number=0, start=0, step=9, n=None):
 PKCOL = [getattr(ImageD11.cImageD11, p) for p in PKSAVE]
 
 
+def collect_all_frames_peaks(
+    master_file,
+    scans,
+    omega_angles,
+    detector,
+    worker_args,
+    num_cpus=None,
+    **process_map_kwargs,
+):
+    """
+    Collects peaks for all the frames in the first scan in the dataset 
+    using parallel processing.
+    """
+    num_threads = num_cpus or max(1, ImageD11.cImageD11.cores_available() - 1)
+    scan_name = scans[0]
+    frames_dataset = f"{scan_name}/measurement/{detector}"
```

Comment:
Maybe I missed something but `scans` and `detector` seem to be used only to compute `frames_dataset`.

Meaning you could pass `frames_dataset` instead of `scans` and `detector`, reducing the number of parameters.

Similarly, only the `omega_angles` that are relevant to the scan could be passed to the function (i.e. pass `omega_angles[0]` instead of the full `omega_angles`.

This would make the function available for any scan, not only the first one if we so desire.